.github/workflows/test.yaml

```diff
@@ -14,7 +14,24 @@ env:
 jobs:
   test:
     runs-on: ubuntu-latest
+
     steps:
     - uses: actions/checkout@v3
-    - name: running tests
-      run: cargo test --verbose
+
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - name: Install Tarpaulin for Coverage
+      run: cargo install cargo-tarpaulin
+
+    - name: Run Tests with Coverage
+      run: cargo tarpaulin --out Html
+
+    - name: Upload Coverage Report
+      uses: actions/upload-artifact@v2
+      with:
+        name: coverage-report
+        path: ./tarpaulin-report.html
```

Comment:
I think the coverage is currently under 10%. would you like to add more tests to increase the coverage too?

Reply:
> I think the coverage is currently under 10%. would you like to add more tests in order to increase the coverage too?

for this i need more time to increase the coverage.so for this i open new pr ASAP((because i need to study more about Rust))

Reply:
> we don't want to upload the coverage. just comparing the results with a fixed percentile would be enough. sth like this:
> 
> ```rust
> $ cargo tarpaulin | tail -n 1 | awk -F% '{print;} $1 + 0 < 10 {exit 1}'
> ```

it's done :)